### DISCUSSION TITLE
Possibility to see the actual value

### DISCUSSION THREAD

**Discussion by jon-coffey** [jamesoff/simplemonitor#1393] [Q&A]
Hi

I just started with simplemonitor in my small prototype env. It works great so far. Great work.

After setting up my monitors I was wondering if I could also receive the actual value that was measured, e.g., I have a diskspace monitor that checks if 20% are at least free. That yield in a result of 1 in the database as the value was true. But can I see somewhere that the diskspace was 13Gb (maybe also measure against the total)? 

This would help if I not only want to have notification but also a dashboard that shows me over-time values of the infrastructure.

Best regards

**Comment by jamesoff** [comment]
Originally I thought I didn't want this when I first started SimpleMonitor but I've been thinking recently having that data available would be useful for (for example) including little sparkline-style graphs in the HTML output would be helpful.

At the moment some monitors report their values as part of the message but that's not consistent. I'll have a think about how to include the data so that it's available for other things to make use of :)

**Reply by jon-coffey** [reply]
What do you think about the idea to leave the monitors as is but introduce observations? Then the og idea is still valid and everybody who just needs a simple yes, no is happy. 

For the observation a value can be written (in the same results table) but is separated as a type of observation. We would then have multiple observation classes, e.g., one for disk size another one for disk used, ... This way every observation would just do one thing but good.

This would also be good because I think that values are only necessary for quantitative results. For a ping or health check I don't see a value.

I just started with python but if there is any chance to collaborate or contribute, I would be happy to do so.

**Reply by jamesoff** [reply]
I think the two can co-exist ok, I was thinking of implementing it just as like a FIFO list of results in the monitor (so it would mostly be handled by code in the `Monitor` base class, and probably add it as an optional field to `record_success()` and `record_fail()` so it can be added in a backward-compatible (and simplicity-compatible) way :)

For pings and health checks you might want to record a latency value so it could still be useful on those, but I agree not all monitors would necessarily need to make use of it.

I'm on holiday at the moment but will have a ponder about how I'd approach it and let you know where you may be able to help — I started simplemonitor as a way to learn Python myself!
